src/mongo/db/pipeline/expression.h

```diff
@@ -1174,6 +1176,9 @@ class ExpressionFilter final : public Expression {
     boost::intrusive_ptr<Expression> _input;
     // The expression determining whether each element should be present in the result array.
     boost::intrusive_ptr<Expression> _filter;
+    // For expression ArrayElemAt and Slice we can optimize filter by setting a limit to end the
```

Comment:
I'd make it clearer that this optimization applies when $filter is being used _as an argument_ to either $slice or $arrayElemAt. Also, feel free to use the $slice name instead of ExpressionSlice here.